crates/vm/src/system/memory/paged_vec.rs

```diff
@@ -372,6 +372,26 @@ impl<const PAGE_SIZE: usize> AddressMap<PAGE_SIZE> {
         self.paged_vecs.iter().all(|page| page.is_empty())
     }
 
+    /// Copies `data` into the memory at `(addr_space, ptr)`.
+    /// # Safety
+    /// - `T` **must** be the correct type for a single memory cell for `addr_space`
+    /// - Assumes `addr_space` is within the configured memory and not out of bounds
+    pub fn copy_slice_nonoverlapping<T: Copy>(&mut self, (addr_space, ptr): Address, data: &[T]) {
+        let start = (ptr as usize) * size_of::<T>();
+        let len = data.len() * size_of::<T>();
+        let mut dst = Vec::with_capacity(len);
+        unsafe {
+            self.paged_vecs
+                .get_unchecked_mut((addr_space - self.as_offset) as usize)
+                .set_range_generic(
```

Comment:
two issues: remember `set_range_generic` has a problem that it fails if the range length is larger than two pages (4096 bytes each). 
it's wasteful to have to allocate to `dst` when you're just going to throw it away.

you can update the implementation of `set_range_generic` itself or introduce a new function to resolve this